### PR TITLE
Add activity feedback button

### PR DESCRIPTION
**Pull request by knowuh**
This PR relates to a series of features required for providing activity level feedback, as per this PT epic: https://www.pivotaltracker.com/epic/show/3777289

Additionally, this PR includes:
  * Not displaying iFramed content in feedback items inline.
  * More visible pop-out links to view students work.
  * Layout adjustments.

@pjanik  Sorry for yet again letting this all pile up in one enormous PR.  If you can't review I can get Doug or someone else have a look.

### CODE REVIEW

**Comment by pjanik** [comment]
No worries, unless there's anyone familiar with the new features, I guess I'm the right person as at least I used to know the code. 😉 I'll take a look at today or tomorrow.

Any chances to play with it? Some deployed demo / example?

**Comment by knowuh** [comment]
Yes,  you can demo with fake data at  http://portal-report.concord.org/branch/add-activity-feedback-button/

Also http://learn.staging.concord.org  is configured to use that branch for reports now too.  You could also ask @eireland on slack about how she QA'd this.  She might have a setup you could borrow.

File: js/components/student-report-link.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  render() {
+    const {student, started} = this.props
+    const student_id = student.get('id')
+    const name = student.get('name')
+    const link = studentReporturl(student_id)
+    const linkToWork =
+      <a href={link} target="_blank">
+        Open {name}'s report
+      </a>
```
Comment (pjanik) [inline js/components/student-report-link.js]: Github diff is unhappy about this syntax, but I guess it works in the browser (right?). ;) Usually people put `( )` around the element or opening tag on the same line.

**Comment by pjanik** [comment]
I played with it and it works fine for me. There's some console warning that I'd fix if possible:
screen shot 2017-11-08 at 13 21 35
But it also shows when I click the old feedback button, so might not be related to those changes.
I also had weird issues with feedback text area, but that got fixed when I disabled one of the chrome plugins.

I read pretty much the whole code. I won't complain about things that can be fixed by linter or prettier. ;) I focused on reducers and actions, and JSON response transformation, and that seems pretty clear. Components are more complicated, I can't say I can follow what that happens there just by reading this huge PR. But they seem to work in practice. :wink:
E.g. I noticed that slim activity component got "connected" to the state now - it was easier to get feedback data like this, right?

In general, it looks good to me. 👍 I'd add standard-js (or prettier) someday, but it's not part of this work. Also, it would be nice to avoid new red warnings in a console if possible, but it might be a separate bug.

**Comment by knowuh** [comment]
Thank you for the review @pjanik --  I will let this build, and then merge if it looks ok.

**Pull request merged**